src/lib/lifeCycle.ts

```diff
@@ -138,15 +138,14 @@ export async function initCli<
     );
   }
 
-  const [userConfig, maybeFluenceConfig] = await Promise.all([
-    ensureUserConfig(),
+  const [maybeFluenceConfig] = await Promise.all([
     initFluenceConfig(),
+    ensureUserConfig(),
   ]);
```

Comment:
I can leave a comment. I use Promise.all to just batch the functions that don't depend on each other. ensureUserConfig is a bit different from other stuff because if this config doesn't exist - we have to ask the user for countly concent which is an async operation as well. User config acts like a singleton so we do not destructure it in array cause it is available everywhere in the program anyway

Reply:
Added a comment